src/frontend/src/flows/authenticate/index.ts

```diff
@@ -0,0 +1,356 @@
+    <div class="modeContainer">
+      <div class="childContainer">
+        <input
+          class="anchorText"
+          type="text"
+          id="userNumberInput"
+          placeholder="Enter anchor"
+          value="${userNumber !== undefined ? userNumber?.toString() : ""}"
+        />
+        <button id="editAnchorButton">${editIcon}</button>
+      </div>
+      <div id="invalidAnchorMessage" class="error-message-hidden">
+        The Identity Anchor is not valid. Please try again.
+      </div>
+    </div>
+    <button type="button" id="authorizeButton" class="primary">
+      Authorize
+    </button>
+    <div id="registerSection">
+      <button type="button" id="registerButton">
+        Create New Identity Anchor
+      </button>
+    </div>
+    <div class="textLink">
+      <hr />
+      <div class="buttonContainer">
+        <button id="recoverButton" class="linkStyle">Lost access?</button>
+      </div>
+      <div class="buttonContainer">
+        <button type="button" class="linkStyle" id="manageButton">
+          Manage your Identity Anchor
+        </button>
+      </div>
+      <hr />
+    </div>
+    ${navbar}
+  </div>
+  ${footer}`;
+
+export interface AuthSuccess {
+  userNumber: bigint;
+  connection: IIConnection;
+  sendDelegationMessage: () => void;
+}
+
+/**
+ * Shows the authorize application view and returns information about the authenticated user and a callback, which sends
+ * the delegation to the application window. After having received the delegation the application will close the
+ * Internet Identity window.
+ */
+export default async (): Promise<AuthSuccess> => {
+  return new Promise((resolve) => {
+    withLoader(async () => {
+      const authContext = await waitForAuthRequest();
+      if (authContext === null) {
+        // The user has manually navigated to "/#authorize".
+        window.location.hash = "";
+        window.location.reload();
+        return;
+      }
+      const userNumber = getUserNumber();
+      init(authContext, userNumber).then(resolve);
+    });
+  });
+};
+
+const init = (
+  authContext: AuthContext,
+  userNumber?: bigint
+): Promise<AuthSuccess> => {
+  displayPage(authContext.requestOrigin, userNumber);
+  initManagementBtn();
+  initRecovery();
+
+  const authorizeButton = document.getElementById(
+    "authorizeButton"
+  ) as HTMLButtonElement;
+  const userNumberInput = document.getElementById(
+    "userNumberInput"
+  ) as HTMLInputElement;
+  const editAnchorButton = document.getElementById(
+    "editAnchorButton"
+  ) as HTMLButtonElement;
+  const registerSection = document.getElementById(
+    "registerSection"
+  ) as HTMLInputElement;
+
+  userNumberInput.onkeypress = (e) => {
+    if (e.key === "Enter") {
+      // authenticate if user hits enter
+      e.preventDefault();
+      authorizeButton.click();
+    }
+  };
+
+  if (userNumber === undefined) {
+    editAnchorButton.classList.toggle("hidden", true);
```

Comment:
If I understand the doc correctly, using `toggle` with `force = true` or `force = false` is just `add()` and `remove()` respectively?

Reply:
Yes, switched them out for clarity.